### PR TITLE
fix: bigtable-hbase-2.x-hadoop incompatible with hbase-shaded-client 2.x

### LINKED ISSUE #2601: bigtable-hbase-2.x-hadoop incompatible with hbase-shaded-client 2.x
The bigtable-hbase-2.x-hadoop dependency expects hbase-shaded-client 2.x, while the code is compiled against HBase 1.x.

#### Steps to reproduce

  1. Setup a Java project and include the dependency com.google.cloud.bigtable:bigtable-hbase-2.x-hadoop:1.15.0. This dependency transitively depends on org.apache.hbase:hbase-shaded-client:2.2.3
  2. call getClusterMetrics() on a AbstractBigtableAdmin

#### Code example

```java
org.apache.hadoop.hbase.client.Connection connection = createConnection();
org.apache.hadoop.hbase.client.AbstractBigtableAdmin admin = connection.getAdmin().
int size = admin.getClusterMetrics().getLiveServerMetrics().size();
```

#### Stack trace
```
Caused by: java.lang.NoSuchMethodError: org.apache.hadoop.hbase.ClusterStatus: method <init>()V not found
	at org.apache.hadoop.hbase.client.AbstractBigtableAdmin$2.<init>(AbstractBigtableAdmin.java:705)
	at org.apache.hadoop.hbase.client.AbstractBigtableAdmin.getClusterStatus(AbstractBigtableAdmin.java:705)
	at com.google.cloud.bigtable.hbase2_x.BigtableAdmin.getClusterMetrics(BigtableAdmin.java:787)
	at org.apache.hadoop.hbase.client.Admin.getClusterMetrics(Admin.java:1564)
	at com.privitar.agrotera.adapters.cluster.hadoop.hbase.apache.admin.ApacheHBaseAdmin.getNumberOfRegionServersInCluster(ApacheHBaseAdmin.java:43)
```

### PR DESCRIPTION
**Pull request by dmitry-fa**
Fixes #2601

### CODE REVIEW

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAdmin.java [hunk 1]
```diff
@@ -833,4 +835,19 @@ public void updateReplicationPeerConfig(String arg0, ReplicationPeerConfig arg1)
       String s, ReplicationPeerConfig replicationPeerConfig) {
     throw new UnsupportedOperationException("updateReplicationPeerConfigAsync"); // TODO
   }
+
+  @Override
+  public ClusterStatus getClusterStatus() throws IOException {
```
[the remark this answers is not in the document]
Comment (dmitry-fa) [reply bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAdmin.java:840]: I love writing tests.  The problem here, that this class doesn't have any unit tests yet. Start covering with the method which is deprecated and does nothing looks a bit strange... But I can.

[the remark this answers is not in the document]
Comment (dmitry-fa) [reply bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAdmin.java:840]: @kolea2 The tests are ready, please take a look.

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/test/java/com/google/cloud/bigtable/hbase2_x/BigtableAdminTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import org.apache.hadoop.hbase.client.AbstractBigtableConnection;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class BigtableAdminTest {
+
+  @Test
+  public void testGetClusterStatus() throws IOException {
+    // test to verify compatibility between 1x and 2x
+    AbstractBigtableConnection connectionMock;
```
Comment (kolea2) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/test/java/com/google/cloud/bigtable/hbase2_x/BigtableAdminTest.java:37]: nit - I would inline these to the declarations

**Review by kolea2 — COMMENTED** [review]
LGTM with one small comment

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/test/java/com/google/cloud/bigtable/hbase2_x/BigtableAdminTest.java (hunk 2 above)
Comment (dmitry-fa) [reply bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/test/java/com/google/cloud/bigtable/hbase2_x/BigtableAdminTest.java]: Good catch! Fixed, please take a look.

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/test/java/com/google/cloud/bigtable/hbase2_x/BigtableAdminTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
+/*
```
Comment (kolea2) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/test/java/com/google/cloud/bigtable/hbase2_x/BigtableAdminTest.java:1]: header should match the format like here: https://github.com/googleapis/java-bigtable-hbase/blob/master/bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableExtendedConfiguration.java#L1

**Review by kolea2 — CHANGES_REQUESTED** [review]
LGTM after last comment change

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/test/java/com/google/cloud/bigtable/hbase2_x/BigtableAdminTest.java (hunk 3 above)
Comment (dmitry-fa) [reply bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/test/java/com/google/cloud/bigtable/hbase2_x/BigtableAdminTest.java:1]: done!

**Review by kolea2 — APPROVED** [review]

**Comment by kolea2** [comment]
Opened #2785 to track this for 1.x branch as well

**Comment by syucream** [comment]
Any plan to release it?

**Pull request merged**